v2/internal/controllers/crd_networking_dnsresolver_test.go

```diff
@@ -0,0 +1,156 @@
+	subnet := newSubnet(tc, vnet, "10.0.0.0/24")
+	tc.CreateResourceAndWait(subnet)
+	defer tc.DeleteResourceAndWait(subnet)
+
+	inbound := &v1api20220701.DnsResolversInboundEndpoint{
+		ObjectMeta: tc.MakeObjectMeta("inbound"),
+		Spec: v1api20220701.DnsResolvers_InboundEndpoint_Spec{
+			IpConfigurations: []v1api20220701.IpConfiguration{
+				{
+					PrivateIpAllocationMethod: to.Ptr(v1api20220701.IpConfiguration_PrivateIpAllocationMethod_Dynamic),
+					Subnet:                    &v1api20220701.DnsresolverSubResource{Reference: tc.MakeReferenceFromResource(subnet)},
+				},
+			},
+			Location: tc.AzureRegion,
+			Owner:    testcommon.AsOwner(resolver),
+		},
+	}
+
+	tc.CreateResourceAndWait(inbound)
+	tc.Expect(inbound.Status.Id).ToNot(BeNil())
+	armId := *inbound.Status.Id
+
+	old := inbound.DeepCopy()
+	key := "foo"
+	inbound.Spec.Tags = map[string]string{key: "bar"}
+
+	tc.PatchResourceAndWait(old, inbound)
+	tc.Expect(inbound.Status.Tags).To(HaveKey(key))
+
+	tc.DeleteResourceAndWait(inbound)
+
+	// Ensure delete
+	exists, retryAfter, err := tc.AzureClient.HeadByID(tc.Ctx, armId, string(v1api20220701.APIVersion_Value))
+	tc.Expect(err).ToNot(HaveOccurred())
+	tc.Expect(retryAfter).To(BeZero())
+	tc.Expect(exists).To(BeFalse())
+}
+
+func DnsResolver_OutboundEndpoint_CRUD(tc *testcommon.KubePerTestContext, resolver *v1api20220701.DnsResolver, vnet *v1api20201101.VirtualNetwork) {
+	subnet := newSubnet(tc, vnet, "10.225.0.0/28")
+	tc.CreateResourceAndWait(subnet)
+	defer tc.DeleteResourceAndWait(subnet)
+
+	outbound := newDnsResolversOutboundEndpoint(tc, resolver, subnet)
+
+	tc.CreateResourceAndWait(outbound)
+	tc.Expect(outbound.Status.Id).ToNot(BeNil())
+	armId := *outbound.Status.Id
+
+	old := outbound.DeepCopy()
+	key := "foo"
+	outbound.Spec.Tags = map[string]string{key: "bar"}
+
+	tc.PatchResourceAndWait(old, outbound)
+	tc.Expect(outbound.Status.Tags).To(HaveKey(key))
+
+	tc.DeleteResourceAndWait(outbound)
+
+	// Ensure delete
+	exists, retryAfter, err := tc.AzureClient.HeadByID(tc.Ctx, armId, string(v1api20220701.APIVersion_Value))
+	tc.Expect(err).ToNot(HaveOccurred())
+	tc.Expect(retryAfter).To(BeZero())
+	tc.Expect(exists).To(BeFalse())
+}
+
+func newDnsResolversOutboundEndpoint(tc *testcommon.KubePerTestContext, resolver *v1api20220701.DnsResolver, subnet *v1api20201101.VirtualNetworksSubnet) *v1api20220701.DnsResolversOutboundEndpoint {
+	outbound := &v1api20220701.DnsResolversOutboundEndpoint{
+		ObjectMeta: tc.MakeObjectMeta("outbound"),
+		Spec: v1api20220701.DnsResolvers_OutboundEndpoint_Spec{
+			Subnet:   &v1api20220701.DnsresolverSubResource{Reference: tc.MakeReferenceFromResource(subnet)},
+			Location: tc.AzureRegion,
+			Owner:    testcommon.AsOwner(resolver),
+		},
+	}
+	return outbound
+}
+
+func newDnsResolver(tc *testcommon.KubePerTestContext, rg *resources.ResourceGroup, vnet *v1api20201101.VirtualNetwork) *v1api20220701.DnsResolver {
```

Comment:
it honestly might be cleaner in these cases where there is no reuse to just inline this into the test method. It makes it more obvious what the expected configuration of the object is.

Reply:
Its being re-used in test `Test_Networking_ForwardingRuleSet_CRUD` [here](https://github.com/Azure/azure-service-operator/pull/3046/files/b5095867f817877b4b23cceb749794963690e555#diff-9664b94c360335d1cd5ba8b1c82aea98fbddfa5771674dd5ea1219c1aea27886R26)